<a href="https://colab.research.google.com/github/ashikKN/DataAcquisition_CaseStudy_ICT/blob/main/DataAcquisition_CaseStudy.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd

In [ ]:
df = pd.read_csv("/content/student_performance_prediction.csv")

In [ ]:
df.head()

Student ID  Study Hours per Week  Attendance Rate  Previous Grades  \
0     S00001                  12.5              NaN             75.0   
1     S00002                   9.3             95.3             60.6   
2     S00003                  13.2              NaN             64.0   
3     S00004                  17.6             76.8             62.4   
4     S00005                   8.8             89.3             72.7   

  Participation in Extracurricular Activities Parent Education Level Passed  
0                                         Yes                 Master    Yes  
1                                          No            High School     No  
2                                          No              Associate     No  
3                                         Yes               Bachelor     No  
4                                          No                 Master     No

Print the summary statistics of the dataset.

In [ ]:
df.describe()

Study Hours per Week  Attendance Rate  Previous Grades
count          38005.000000     38008.000000     38006.000000
mean               9.962744        75.276323        65.440107
std                5.031154        20.393418        16.503119
min              -12.300000       -14.300000         8.300000
25%                6.600000        61.600000        55.100000
50%               10.000000        75.300000        65.200000
75%               13.400000        88.800000        75.200000
max               32.400000       150.200000       200.000000

1. What is the size of the dataset? (Number of rows and columns)

In [ ]:
rows_column = df.shape

print(f"No: of rows = {rows_column[0]}")
print(f"No: of column = {rows_column[1]}")

No: of rows = 40000
No: of column = 7


2. What are the data types of each column?

In [ ]:
df.dtypes

Student ID                                      object
Study Hours per Week                           float64
Attendance Rate                                float64
Previous Grades                                float64
Participation in Extracurricular Activities     object
Parent Education Level                          object
Passed                                          object
dtype: object

3. Are there any missing values in the dataset? If so, which columns have
missing values and how many?

In [ ]:
df.isnull()

Student ID  Study Hours per Week  Attendance Rate  Previous Grades  \
0           False                 False             True            False   
1           False                 False            False            False   
2           False                 False             True            False   
3           False                 False            False            False   
4           False                 False            False            False   
...           ...                   ...              ...              ...   
39995       False                 False            False            False   
39996       False                 False            False            False   
39997       False                 False            False            False   
39998       False                 False            False             True   
39999       False                 False            False            False   

       Participation in Extracurricular Activities  Parent Education Level  \
0                                            False                   False   
1                                            False                   False   
2                                            False                   False   
3                                            False                   False   
4                                            False                   False   
...                                            ...                     ...   
39995                                        False                   False   
39996                                        False                   False   
39997                                        False                   False   
39998                                        False                   False   
39999                                        False                   False   

       Passed  
0       False  
1       False  
2       False  
3       False  
4       False  
...       ...  
39995   False  
39996   False  
39997   False  
39998   False  
39999   False  

[40000 rows x 7 columns]

4. What are the unique values in a categorical column of your choice? How
many unique values are there?

In [ ]:
unique_values = df['Parent Education Level'].unique()
print(unique_values)

count = df['Parent Education Level'].nunique()
print(f"\n No of unique values in Parent Education Level : {count}")

['Master' 'High School' 'Associate' 'Bachelor' 'Doctorate' nan]

 No of unique values in Parent Education Level : 5


5. For a numerical column of your choice, calculate the mean, median, and
standard deviation.

In [ ]:
mean = df['Study Hours per Week'].mean()
print(f"Mean Value for Study Hours Per Week = {mean}")

median = df['Study Hours per Week'].median()
print(f"Median Value for Study Hours Per Week = {median}")

std_dev = df['Study Hours per Week'].std()
print(f"Standard Deviation for Study Hours Per Week = {std_dev}")

Mean Value for Study Hours Per Week = 9.962744375740034
Median Value for Study Hours Per Week = 10.0
Standard Deviation for Study Hours Per Week = 5.031153759403836


6. Identify any potential outliers in a numerical column of your choice. Explain
your approach.

In [ ]:
column = 'Study Hours per Week'
Q1 = df[column].quantile(0.25)
Q3 = df[column].quantile(0.75)
IQR = Q3-Q1
lower_bound = Q1-1.5*IQR
upper_bound = Q3+1.5*IQR
outliers = df[(df[column] < lower_bound) | (df[column] > upper_bound)]
outliers

Student ID  Study Hours per Week  Attendance Rate  Previous Grades  \
209       S00210                  29.3              NaN             55.1   
262       S00263                  -6.2             98.7             26.2   
478       S00479                  25.4             84.5             53.0   
1061      S01062                  -4.2             70.8             46.3   
1539      S01540                  -4.4             77.7             54.0   
...          ...                   ...              ...              ...   
39499     S39500                  -4.2             57.0             33.3   
39527     S39528                  -4.5             66.9             73.7   
39575     S39576                  24.3             61.8             48.0   
39616     S39617                  -7.2             97.8             46.3   
39838     S39839                  26.5              NaN             69.3   

      Participation in Extracurricular Activities Parent Education Level  \
209                                           Yes               Bachelor   
262                                           Yes            High School   
478                                            No            High School   
1061                                           No                 Master   
1539                                           No               Bachelor   
...                                           ...                    ...   
39499                                         NaN                 Master   
39527                                          No            High School   
39575                                          No              Associate   
39616                                         Yes            High School   
39838                                         Yes              Associate   

      Passed  
209      Yes  
262      Yes  
478      Yes  
1061      No  
1539      No  
...      ...  
39499     No  
39527     No  
39575     No  
39616     No  
39838    Yes  

[251 rows x 7 columns]

7. Find the correlation between numerical columns and discuss any interesting
relationships.

In [ ]:
numerical_df = df.select_dtypes(include=['int64', 'float64'])
correlation_matrix = numerical_df.corr()
print(correlation_matrix)

                      Study Hours per Week  Attendance Rate  Previous Grades
Study Hours per Week              1.000000         0.004654         0.000587
Attendance Rate                   0.004654         1.000000        -0.005034
Previous Grades                   0.000587        -0.005034         1.000000


8. Group the data by a categorical column and compute the aggregate statistics
(e.g., mean, sum) for a numerical column.

In [ ]:
aggregate_stats = df.groupby('Participation in Extracurricular Activities').agg({
    'Study Hours per Week': ['mean', 'sum'],
    'Attendance Rate': ['mean', 'sum'],
    'Previous Grades': ['mean', 'sum']
})

print(aggregate_stats)

                                            Study Hours per Week            \
                                                            mean       sum   
Participation in Extracurricular Activities                                  
No                                                      9.921882  179645.6   
Yes                                                     9.999889  180148.0   

                                            Attendance Rate             \
                                                       mean        sum   
Participation in Extracurricular Activities                              
No                                                75.246814  1362569.3   
Yes                                               75.349425  1355611.5   

                                            Previous Grades             
                                                       mean        sum  
Participation in Extracurricular Activities                             
No                 

9. **Based on your analysis, provide a brief summary of any insights or patterns
you discovered in the dataset.**

Based on the analysis of the dataset, here are some key insights and patterns :

General Overview :
----------------
Study Hours per Week:
1. The average number of study hours per week is approximately 9.96 hours.
2.The median study hours per week is 10.0 hours, which is close to the mean, indicating a fairly symmetric distribution of study hours.
3.The standard deviation is 5.03 hours, suggesting moderate variability in study hours among students.

Attendance Rate:
The average attendance rate is about 75.28%.
There is a wide range of attendance rates, with a standard deviation of 20.39%, indicating significant variability in student attendance.
Previous Grades:

Previous Grades:
The average previous grade is 65.44, with a standard deviation of 16.50, indicating some variability in academic performance.

Extreme Values and Potential Data Issues:
-----------------------------------------
There are negative values in Study Hours per Week (-12.3) and Attendance Rate (-14.3), which are not logically possible and suggest data entry errors or outliers. These should be investigated and cleaned from the dataset.

The maximum values for Attendance Rate (150.2%) and Previous Grades (200.0) also exceed logical limits, further indicating potential data issues.

Correlation Between Numerical Variables
---------------------------------------
The correlation coefficients between Study Hours per Week, Attendance Rate, and Previous Grades are all very close to zero, indicating no significant linear relationships between these variables.

Conclusions:
------------
Data Quality Concerns: The presence of extreme and illogical values (negative and excessively high numbers) indicates the need for data cleaning before making further inferences.

Minimal Impact of Extracurricular Activities: Participation in extracurricular activities does not significantly alter study habits, attendance, or grades, though slight positive differences are observed.

No Strong Linear Relationships: The weak correlations suggest that other factors might be more significant in influencing study hours, attendance, and grades, and these relationships may require more complex modeling or consideration of other variables.

